feat: implement max metric aggregation

hamzahejja opened the pull request:
- This PR implements missing `Max` metric aggregation, Reference Issue: #20 
- Add tests to cover the implementation.

hamzahejja committed:
feat: implement max metric aggregation

hamzahejja committed:
add docblock to reference added max aggregation method

hamzahejja committed:
change accessibility modifier

Douglasdc3 commented on src/Query/Aggregations/Metric/Max.php:
```diff
@@ -0,0 +1,28 @@
+     */
+    private $field;
+
+    public function __construct(string $field)
```
The Max aggregation also accepts an optional `missing` argument used when no value is found in elastic. 

https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-max-aggregation.html#_missing_value_4

hamzahejja replied:
@Douglasdc3 The reason I did NOT add it is that we didn't do it in the **Min** aggregation although it does have the same option. Should I go back and include it in the **Min** aggregation as well ?

Moreover, looking into reference doc(s) for the `Cardinality` aggregation which we implemented earlier, looks like it has a `precision_threshold` optional option that we can as well, but still forgot to add it on. Should I go back and include this one as well ? 

[Min Aggregation](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-min-aggregation.html)

[Cardinality Aggregation](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-metrics-cardinality-aggregation.html)

hamzahejja committed:
add optional  parameter to Max aggregation constructor and cover it within test

add doc-block for max aggregation with passing  parameter

add optional  parameter for Min aggregation and modify tests and add necessary docblock

add optional parameters for Cardinality metric aggregation and extend test

code style: PHP-CS-Fixer

Better enhanced/neat implementation for optional parameters of metric aggregations

hamzahejja committed:
feat: filter-through only allowed/available additional optional parameters for aggs

Douglasdc3 commented:
Thanks for the addition! 

Few small things cleaned up in new commit on master.
